app/oess_pull_azure_interfaces.pl

```diff
@@ -88,6 +113,70 @@ sub reconcile_oess_endpoints {
     }
 }
 
+sub find_matching_azure_connection{
+    my $azure_connection_info = shift;
+    my $cloud_interconnect_id = shift;
+    my $azure_peering_subnet;
+    my $peering = $azure_connection_info->{properties}->{peerings};
+    foreach my $ip (@$peering){
+        if($cloud_interconnect_id =~ m/-SEC-/){
+            $azure_peering_subnet = $ip->{properties}->{secondaryPeerAddresssubnet};
+        }elsif($cloud_interconnect_id =~ m/-PRI-/){
+            $azure_peering_subnet = $ip->{properties}->{primaryPeerAddresssubnet};
+        }else{
+            $azure_peering_subnet = "";
+        }
+    }
+    return $azure_peering_subnet;
+}
+
+sub get_endpoint_peer{
+    my $endpoint = shift;
+    my $cloud_interconnect_id = shift;
+    my $cloud_account_id = shift;
+    my $peers = $endpoint->peers();
+    my $endpoint_ips =[];
+    if($endpoint->cloud_interconnect_id() eq $cloud_interconnect_id && $endpoint->cloud_account_id() eq $cloud_account_id){
+        foreach my $peer (@$peers){
+            my $cloud_connetions = $peer->{db}->{configuration}->{cloud}->{connection};
+            foreach my $conn (@$cloud_connetions){
+                if( $conn->{interconnect_id} eq $cloud_interconnect_id){
+                    return $peer;
+                } 
+            }
+        }
+    }
+    return undef;
+}
+
+sub update_endpoint_peer_ips{
+    my $db = shift;
+    my $azure_subnet = shift;
+    my $peer = shift;
+    my $new_oess_ip = increment_ip($azure_subnet, 1);
+    my $new_azure_ip = increment_ip($azure_subnet, 2);
+    my $logger = Log::Log4perl->get_logger('OESS.Cloud.Azure.Syncer');
+    $logger->info("UPDATEING AZURE PEERING: changing $peer->{local_ip} to $new_oess_ip, and changing $peer->{peer_ip} to $new_azure_ip");
+    $peer->{local_ip} = $new_oess_ip;
+    $peer->{peer_ip} = $new_azure_ip;
+    $peer->update($db, $peer->to_hash());
```

Comment:
`$peer->update` doesn't take any arguments. You may simply this call to `$peer->update;`.

Reply:
changed $peer->update($db, $peer->to_hash()); to $peer->update;